src/dune_pkg/rev_store.ml

```diff
@@ -297,8 +303,49 @@ module Remote = struct
   ;;
 end
 
+module Source = struct
```

Comment:
If this type is necessary, it should probably live in the `Opam_repo`. It's not really modelling a property of the revision store as it doesn't care about the association between remotes and commits

Reply:
I would disagree, in my opinion `Source.t` represents a thing that you can add to the revision store and as the store is backed by Git, it needs to know how to represent things in the revision store and not in the `Opam_repo`. 

One might disagree that the commit that is chosen is not really a feature of the source but rather a choice on what revision to check out, though if the user picks a branch we might want to use that branch as the `default_branch`.

Also, it can't be added to `Opam_repo` as that would create a cyclic dependency, so it would need to live in its own place.